momepy/preprocessing.py

```diff
@@ -740,3 +746,331 @@ def _get_extrapolated_line(coords, tolerance, point=False):
+
+    # only keeping relevant fields
+    geom_col = rab_plus.geometry.name
+    rab_plus = rab_plus[[geom_col, "index_right"]]
+
+    return rab_plus
+
+
+def _rabs_center_points(gdf, center_type="centroid"):
+    """
+    From a selection of roundabouts, returns an aggregated GeoDataFrame
+    per round about with extra column with center_type.
+    """
+    # creating a multipolygon per RAB (as opposed to dissolving) of the entire
+    # composition of the RAB
+    # temporary DataFrame where geometry is the array of pygeos geometries
+    tmp = pd.DataFrame(gdf.copy())  # temporary hack until shapely 2.0 is out
+    tmp["geometry"] = tmp.geometry.values.data
+
+    pygeos_geoms = (
+        tmp.groupby("index_right")
+        .geometry.apply(pygeos.multipolygons)
+        .rename("geometry")
+    )
+    pygeos_geoms = pygeos.make_valid(pygeos_geoms)
+
+    rab_multipolygons = gpd.GeoDataFrame(pygeos_geoms, crs=gdf.crs)
+    # make_valid is transforming the multipolygons into geometry collections because of
+    # shared edges
+
+    if center_type == "centroid":
+        # geometry centroid of the actual circle
+        rab_multipolygons["center_pt"] = gdf[
+            gdf.index == gdf.index_right
+        ].geometry.centroid
+
+    elif center_type == "mean":
+        coords, idxs = pygeos.get_coordinates(pygeos_geoms, return_index=True)
+        means = {}
+        for i in np.unique(idxs):
+            tmps = coords[idxs == i]
+            target_idx = rab_multipolygons.index[i]
+            means[target_idx] = Point(tmps.mean(axis=0))
+
+        rab_multipolygons["center_pt"] = gpd.GeoSeries(means, crs=gdf.crs)
+
+    # centerpoint of minimum_bounding_circle
+    # minimun_bounding_circle() should be available in Shapely 2.0. Implementation still
+    # pending.
+    # current environment has 1.8.2
+
+    return rab_multipolygons
+
+
+def _coins_filtering_many_incoming(incoming_many, angle_threshold=0):
+    """
+    Used only for the cases when more than one incoming line touches the
+    roundabout.
+    """
+    coins_filter_result = []
+    # For each new connection, evaluate COINS and select the group from which the new
+    # line belongs
+    # TODO ideally use the groupby object on line_wkt used earlier
+    for g, x in incoming_many.groupby("line_wkt"):
+        gs = gpd.GeoSeries(pd.concat([x.geometry, x.line]), crs=incoming_many.crs)
+        gdf = gpd.GeoDataFrame(geometry=gs)
+        gdf.drop_duplicates(inplace=True)
+
+        coins = COINS(gdf, angle_threshold=angle_threshold)
+        # coins.stroke_attribute()) # the groups here don't match the stroke_group in
+        # .stroke_gdf()
+        stroke_gdf = coins.stroke_gdf()
+        if GPD_10:
+            orig_geom_join = stroke_gdf.sjoin(
+                gpd.GeoDataFrame(geometry=x.line), predicate="covers"
+            )
+        else:
+            orig_geom_join = stroke_gdf.sjoin(
+                gpd.GeoDataFrame(geometry=x.line), op="covers"
+            )
```

Comment:
```suggestion
            orig_geom_join = gpd.sjoin(stroke_gdf,
                gpd.GeoDataFrame(geometry=x.line), op="covers"
            )
```
You need to use a function for older geopandas, the method is newer. Also note that this suggestion does not follow `black`.